### DISCUSSION TITLE
Compatibility with old glibc

### DISCUSSION THREAD

**Discussion by brendene** [odygrd/quill#265] [General]
Hello,

Thanks again for the great software.  On older systems with glibc < 2.17, `std::chrono::steady_clock::now()` is implemented as a syscall in libstdc++ instead of using the faster VDSO (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=59177)

With the default logger config, the logging thread consumes close to 100% cpu due to the syscalls in an almost tight loop.  Setting `backend_thread_strict_log_timestamp_order=false` will stop the `clock_gettime` syscalls in the logging thread or setting `backend_thread_sleep_duration` will reduce the frequency of calls.

FWIW, I came across this when compiling quill in an environment created with conda forge, the version of glibc that their libstdc++ library uses is old, perhaps for compatibility reasons.

I am not sure if its the responsibility of your program to handle these types but that I would point out that they exist and are not uncommon.

**Comment by odygrd** [comment]
hey, thanks for the post :) I haven't tried it in such old systems. 
Yes even by design the backend logging thread spins instead of waking up via a condition variable for example as it is more performant for the hot threads. The logging thread can be pinned to a non-critical core to reduce the effect. 

As you said setting `backend_thread_strict_log_timestamp_order=false` will greatly reduce calls to `std::chrono::steady_clock::now()` I think that is the main one

You can also set (rdtsc_resync_interval) value a bit higher like `900 ms` (https://github.com/odygrd/quill/blob/master/quill/include/quill/Config.h#L178) although it should make only minor difference
